Lib/python/pyhead.swg

```diff
@@ -43,8 +43,11 @@ SWIG_Python_str_AsChar(PyObject *str)
   if (str) {
     char *cstr;
     Py_ssize_t len;
-    PyBytes_AsStringAndSize(str, &cstr, &len);
+    if (PyBytes_AsStringAndSize(str, &cstr, &len) == -1)
+      return NULL;
     newstr = (char *) malloc(len+1);
+    if (!newstr)
+      return NULL;
```

Comment:
Please change to return only at the end of the function to void a leak due to `Py_XDECREF(str)` possibly not being called.